pkg/resource/edit/operations_test.go

```diff
@@ -158,6 +158,41 @@ func TestDeletingDuplicateURNs(t *testing.T) {
 	})
 }
 
+func TestDeletingDuplicateProviderURN(t *testing.T) {
+	t.Parallel()
+
+	// Create duplicate provider resources
+	pA := NewProviderResource("a", "p1", "0")
+	pB := NewProviderResource("a", "p1", "1")
+
+	// Create a resource that depends on the duplicate Provider.
+	b1 := NewResource("b", pA)
+	b2 := NewResource("b", pB)
+
+	c := NewResource("c", pB, b1.URN)
+
+	t.Run("do-target-dependents", func(t *testing.T) {
+		t.Parallel()
+		snap := NewSnapshot([]*resource.State{
+			pA, pB, b1, b2, c,
+		})
+
+		err := DeleteResource(snap, pA, nil, true /* targetDependents */)
+		require.NoError(t, err)
+	})
+
+	t.Run("do-not-target-dependents", func(t *testing.T) {
+		t.Parallel()
+		snap := NewSnapshot([]*resource.State{
+			pA, pB, b1, b2, c,
+		})
+
+		err := DeleteResource(snap, pA, nil, false /* targetDependents */)
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "Can't delete resource \"urn:pulumi:test::test::pulumi:providers:a::p1\" due to dependent resources", pA.URN)
+	})
+}
+
```

Comment:
Can we add the test cases I describe as actual test cases here.

Reply:
I added a better assert for the first test case that I already had here, and also added the second test case you suggested.